ocfweb/docs/views/servers.py

```diff
@@ -57,81 +57,89 @@ def has_munin(self):
+        verify=ROOT_DIR + 'puppet-ca.pem', params={'query': query},
+    )
+    output = eval(r.text)
+    result = {}
+    for d in output:
+        hostname = d['certname'].split('.')[0]
+        result[hostname] = d.get('value')
+    return result
+
+
+def create_hosts(lst):
+    """Accepts a list of raw ldap output, returns a dictionary of host
+    objects indexed by hostname
+    """
+    hosts = {}
+    for h in lst:
+        if not is_hidden(h):
+            description = h.get('description', [''])[0]
+            hostname = h['cn'][0]
+            hosts[hostname] = (Host(hostname, h['type'], description, ()))
+    return hosts
+
+
+@periodic(300)
 def get_hosts():
-    return [
-        Host.from_ldap(
-            hostname='hal',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'fallingrocks',
-                    'monsoon',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='jaws',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'anthrax',
-                    'biohazard',
-                    'death',
-                    'dementors',
-                    'democracy',
-                    'firestorm',
-                    'flood',
-                    'lightning',
-                    'maelstrom',
-                    'pestilence',
-                    'reaper',
-                    'supernova',
-                    'thunder',
-                    'tsunami',
-                    'werewolves',
-                    'whirlwind',
-                    'whiteout',
-                    'zombies',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
```

Comment:
I consolidated the queries. Please see my latest commit. I am not sure if I should get rid of the function to interpret the output of hosts_by_filter though, since I do need to filter out hidden hosts and convert the LDAP output to `Host` objects.